src/table.c

```diff
@@ -0,0 +1,135 @@
+#include <stdlib.h>
+#include <string.h>
+
+#include "table.h"
+
+/* Função para criar e inicializar uma nova tabela hash, com n
+ * linhas (n = módulo da função hash).
+ * Retorna a tabela ou NULL em caso de erro.
+ */
+struct table_t *table_create(int n){
+    struct table_t *table;
+    table = (struct table_t*) calloc(1,sizeof(struct table_t));
+    table->size = n;
+    table->lists = (struct list_t**) calloc(n,sizeof(struct list_t*)); 
+    return table;
+}
+
+/* Função que elimina uma tabela, libertando *toda* a memória utilizada
+ * pela tabela.
+ * Retorna 0 (OK) ou -1 em caso de erro.
+ */
+int table_destroy(struct table_t *table){
+    int status = 0;
+    for(int i=0; i<table->size; i++){
+        status = list_destroy(table->lists[i]);
+        if (status != 0) return status;
+    }
+    return status;
+
+}
+
+/* Função que calcula o índice da lista a partir da chave
+ */
+int hash_code(char *key, int n){
+    //does something, e.g.
+    int hashed = strlen(key);
+    //we can workshop a better hash code to use
+
+    return hashed%n;
+}
+
+/* Função para adicionar um par chave-valor à tabela. Os dados de entrada
+ * desta função deverão ser copiados, ou seja, a função vai criar uma nova
+ * entry com *CÓPIAS* da key (string) e dos dados. Se a key já existir na
+ * tabela, a função tem de substituir a entry existente na tabela pela
+ * nova, fazendo a necessária gestão da memória.
+ * Retorna 0 (ok) ou -1 em caso de erro.
+ */
+int table_put(struct table_t *table, char *key, struct data_t *value){
```

Comment:
We are lacking the check for errors in all functions I think, as here -1 is returned in case of error and in some other functions it's to be returned NULL in case of error

Reply:
I'll double check but for most of these, but there are no extra checks because they return the same NULL/-1 value that list does